src/hotspot/share/services/heapDumper.cpp

```diff
@@ -1918,7 +1931,8 @@ void HeapObjectDumper::do_object(oop o) {
   if (o->is_instance()) {
     // create a HPROF_GC_INSTANCE record for each object
     DumperSupport::dump_instance(writer(), o, &_class_cache);
-    if (java_lang_VirtualThread::is_instance(o) && ThreadDumper::should_dump_vthread(o)) {
+    if (java_lang_VirtualThread::is_instance(o)
```

Comment:
Suggestion. Just to be clear add a comment
```
// If we encounter an unmounted virtual thread it needs to be dumped explicitly.
```
or something to that effect. Thanks.

Reply:
Thank you for review
Added comment as suggested